lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -86,37 +92,55 @@ def configure(conf)
         end
       end
 
-      unless @use_metadata_service
-        unless @project_id && @zone && @vm_id
-          raise Fluent::ConfigError,
-              ('Please specify "project_id", "zone" and "vm_id" if you set "use_metadata_service" to false')
+      if @use_metadata_service
+        if @project_id.nil?
+          begin
+            @project_id = fetch_metadata('project/project-id')
+          rescue
```

Comment:
let me restructure it as I described, the new scheme will be:

- idetermine the platform we're running on by consulting the metadata service (currently this will be either 'GCE' or 'NONE')
- get rid of the 'rescue' clauses for other fetch_metadata calls.
- inside fetch_metadata, if platform == NONE, I'll either return nil or throw an exception (let me think about that).